### DISCUSSION TITLE
EcoTec Plus VC20CS/1-5 PrEnergy* problem

### DISCUSSION THREAD

**Discussion by iszczyg** [john30/ebusd#985] [Q&A]
Hello,
I've installed ebus adapter v.5 in Vaillant EcoTec boiler. It works perfectly. However, I have a question regarding the configuration (I was browsing the forum but without success). Almost all registers are read correctly. Unfortunately one group (PrEnergy*) returns the error (ERR: invalid position for...). I have found the sugestion, that my boiler has no such sensors. But I can read the energy consumption on the boiler lcd panel so I suspect that it has the sensors. Do you know what can be a reason for this? Thanks in advance for your help. I've planned to control the gas consumption with ebusd so this registers are quite important for me. Ebusd reports the bai.308523.inc config file
The output of ebusctl -find:
...
bai PowerValue =  (ERR: invalid position for 3108b509030daa00 / 00)
bai PrAPSCounter =  (ERR: invalid position for 3108b509030df200 / 00)
bai PrAPSSum =  (ERR: invalid position for 3108b509030df300 / 00)
bai PrEnergyCountHc1 =  (ERR: invalid position for 3108b509030df600 / 00)
bai PrEnergyCountHc2 =  (ERR: invalid position for 3108b509030df800 / 00)
bai PrEnergyCountHc3 =  (ERR: invalid position for 3108b509030dfa00 / 00)
bai PrEnergyCountHwc1 =  (ERR: invalid position for 3108b509030dc600 / 00)
bai PrEnergyCountHwc2 =  (ERR: invalid position for 3108b509030dc800 / 00)
bai PrEnergyCountHwc3 =  (ERR: invalid position for 3108b509030dca00 / 00)
bai PrEnergySumHc1 =  (ERR: invalid position for 3108b509030df500 / 00)
bai PrEnergySumHc2 =  (ERR: invalid position for 3108b509030df700 / 00)
bai PrEnergySumHc3 =  (ERR: invalid position for 3108b509030df900 / 00)
bai PrEnergySumHwc1 =  (ERR: invalid position for 3108b509030dc500 / 00)
bai PrEnergySumHwc2 =  (ERR: invalid position for 3108b509030dc700 / 00)
bai PrEnergySumHwc3 =  (ERR: invalid position for 3108b509030dc900 / 00)
bai ProductionByte =  (ERR: invalid position for 3108b509030d3e04 / 00)
...
scan id = no data stored
scan.08  = Vaillant;BAI00;0807;7603
scan.08 id = 21;22;31;0010024647;0001;010956;N7
scan.15  = Vaillant;CTLV2;0514;1104
scan.15 id = 21;22;13;0020260916;0953;016134;N8

**Comment by Dominik-1980** [comment]
@iszczyg 

Hello, I have the exact same problem. Did you found out any solution?

Greetings

**Reply by iszczyg** [reply]
Hello,
Mentioned registers don't  work. They probably do not exist - don't know why. However I've found workaround. Except of boiler I have also controller VRC 720/2. In the configuration directory (ebusd-configuration/latest/en/vaillant) I've created link 15.ctlv2.csv -> 15.700.csv. Now "ebusctl find" shows:  
scan.08  = Vaillant;BAI00;0807;7603
scan.08 id = 21;22;31;0010024647;0001;010956;N7
scan.15  = Vaillant;CTLV2;0514;1104
scan.15 id = 21;22;13;0020260916;0953;016134;N8
Device 15 is VRC 720. There are a set of working registers corresponding to fuel and energy consumption:
            'prEnergySumHc',
            'prEnergySumHcLastMonth',
            'prEnergySumHcThisMonth',
            'prEnergySumHwc',
            'prEnergySumHwcLastMonth',
            'prEnergySumHwcThisMonth',
            'prFuelSumHc',
            'prFuelSumHcLastMonth',
            'prFuelSumHcThisMonth',
            'prFuelSumHwc',
            'prFuelSumHwcLastMonth',
            'prFuelSumHwcThisMonth',
The readout is in strange units but half year experience lead me to the coefficient which can be used to calculate m3 of gas consumption from the readouts: in my case it is 0.1034. The error is below 2%

Best regards

**Comment by Dominik-1980** [comment]
Thanks for your quick reply!
Great you found a workaround!

I will try this later.

**Comment by Dominik-1980** [comment]
Okay, you are reading these values from your smart thermostat. 
I don't have such a device. I use tado, and tado has very little values it handles and no energy consumption over ebus.
Thanks you anyway!
I think the values are stored somewhere in bai, too. They just don't decode the right way...

**Reply by madison2** [reply]
@iszczyg @Dominik-1980 Have you found any solution? I have the same problem.
I have the same errors:
bai PowerValue = (ERR: invalid position for 3108b509030daa00 / 00)
bai PrAPSCounter = (ERR: invalid position for 3108b509030df200 / 00)
bai PrAPSSum = (ERR: invalid position for 3108b509030df300 / 00)
bai PrEnergyCountHc1 = (ERR: invalid position for 3108b509030df600 / 00)
bai PrEnergyCountHc2 = (ERR: invalid position for 3108b509030df800 / 00)
bai PrEnergyCountHc3 = (ERR: invalid position for 3108b509030dfa00 / 00)
bai PrEnergyCountHwc1 = (ERR: invalid position for 3108b509030dc600 / 00)
bai PrEnergyCountHwc2 = (ERR: invalid position for 3108b509030dc800 / 00)
bai PrEnergyCountHwc3 = (ERR: invalid position for 3108b509030dca00 / 00)
bai PrEnergySumHc1 = (ERR: invalid position for 3108b509030df500 / 00)
bai PrEnergySumHc2 = (ERR: invalid position for 3108b509030df700 / 00)
bai PrEnergySumHc3 = (ERR: invalid position for 3108b509030df900 / 00)
bai PrEnergySumHwc1 = (ERR: invalid position for 3108b509030dc500 / 00)
bai PrEnergySumHwc2 = (ERR: invalid position for 3108b509030dc700 / 00)
bai PrEnergySumHwc3 = (ERR: invalid position for 3108b509030dc900 / 00)
bai ProductionByte = (ERR: invalid position for 3108b509030d3e04 / 00)

How you resolve it? Could I please provide instructions on how to make link 15.ctlv2.csv -> 15.700.csv ?
